### PR TITLE
feat(api): add themes endpoint

### PR DESCRIPTION
**Pull request by lionelB**
add themes endpoint to api
`/api/v1/themes/41-licenciement-pour-motif-personnel`

will return 
```
{
  theme: {id: 41, title: "Licenciement pour motif personnel", slug: "41-licenciement-pour-motif-personnel" },
  breadcrumbs: [{}], // an array containing the parent themes 
  children: [{}] // an array containing the children theme
}
```

### CODE REVIEW

File: packages/code-du-travail-data/dataset/themes/query.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const themes = require("./themes.json");
+const memoizee = require("memoizee");
```
Comment (UnbearableBear) [inline packages/code-du-travail-data/dataset/themes/query.js:2]: I would have called it `memoize` but I don't care ^^

Comment (lionelB) [reply packages/code-du-travail-data/dataset/themes/query.js:2]: on the other hand we know that's the memoize function from a 3rd party lib ;)

File: packages/code-du-travail-data/dataset/themes/index.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
+const cstojson = require("csvtojson");
+const slugify = require("slugify");
+
+/**
+ *
+ * flattenThemes take a csv file of themes
```
Comment (UnbearableBear) [inline packages/code-du-travail-data/dataset/themes/index.js:6]: Very nice to sample some data here

File: packages/code-du-travail-data/dataset/themes/index.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+ * }, {
+ *    niveau1: '1 - Recrutement et contrat de travail',
+ *    niveau2: '9 - Recrutement',
+ *    niveau3: '51 - Curriculum vitae (CV)',
+ *    niveau4: ''
+ * }]
+ *
+ *
+ *  and will return a flat list of theme
+ * {
```
Comment (UnbearableBear) [inline packages/code-du-travail-data/dataset/themes/index.js]: Does the output here matches the sample above ?? I'm not sure I get the logic if so

Comment (lionelB) [reply packages/code-du-travail-data/dataset/themes/index.js]: missing 50 and 51 in the output

Comment (UnbearableBear) [reply packages/code-du-travail-data/dataset/themes/index.js]: Ok got it !

File: packages/code-du-travail-data/dataset/themes/index.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+/**
+ * Extract a slug from a chained theme breadcrumbs
+ * input : 46 - Conflits au travail et contrôle de la réglementation - Sanctions disciplinaires
+ * output : 46-sanctions-disciplinaires
+ * @param {String} str complete str
+ * @returns {String} a slugify version of the id+theme
+ */
+
+function extractSlug(str) {
```
Comment (UnbearableBear) [inline packages/code-du-travail-data/dataset/themes/index.js:91]: Ie this one used somewhere ?

**Review by UnbearableBear — APPROVED** [review]
Nice !!

File: packages/code-du-travail-data/dataset/themes/index.js (hunk 4 above)
Comment (lionelB) [reply packages/code-du-travail-data/dataset/themes/index.js:91]: yes the function is exported to be used in `cdtn-theme-{sp|mt}.js` scripts

**Pull request merged**